api/types_test.go

```diff
@@ -0,0 +1,143 @@
+package api
+
+import (
+	"os"
+	"testing"
+)
+
+func TestOIDCConfig_Validate(t *testing.T) {
+	OIDCNotEnabled := false
+	OIDCIsEnabled := true
+	emptyFilename := ""
+	filenameThatDoesntExist := "doesnt-exist.yaml"
```

Comment:
A better alternative that doesn't require setting up the files all at the beginning is to add have an attribute in the `fields` struct that is of type `func() string` that returns the file name. In that function you would add the logic to create the file and such and then return the name of the filename.

Then on the loop you would call that function and from the returned result you would also perform a deferred call to os.Remove using that returned file name to perform the cleanup

Reply:
Also remember my https://github.com/bf2fc6cc711aee1a0c2a/observability-remote-write-proxy/pull/23#discussion_r1133701934 comment. The reason this test is this complicated is because the file reading is coupled to the validation. This should be improved, but that can be dealt with in another PR.